airbyte-integrations/connectors/source-bing-ads/source_bing_ads/spec.json

```diff
@@ -84,6 +86,71 @@
       "user_id": {
         "type": "string",
         "description": "Unique user identifier."
+      },
+      "reports_start_date": {
+        "type": "string",
+        "format": "date",
+        "default": "2020-01-01",
+        "description": "From which date perform initial sync for report related streams. In YYYY-MM-DD format"
+      },
+      "report_aggregation": {
```

Comment:
need clarification:

So now user should choose one of the aggregation types,  

Did we consider choosing a few types at the same time - hourly and monthly for example.

Do you have any ideas on how it works in other EL tools?

Reply:
**Did we consider choosing a few types at the same time - hourly and monthly for example.**
no, I don't consider this. What's the purpose of that ?
**Do you have any ideas on how it works in other EL tools?**
one tool has hardcoded Daily value, another tools has 2 separate streams for Daily and Hourly aggregation. I think my solution is more flexible

Reply:
Please raise this question on Airbyte review: "Did we consider choosing a few types at the same time - hourly and monthly for example"

Reply:
In my understanding, we should support sync of hourly, daily or another 'agreggations' streams simultaneously.

We may also discuss this question with Maxim before Airbyte review.

Reply:
I think we can offer flexibility here pretty easily on top of the current code and therefore imo we should.

Options in spec could be boolean toggles for each of `hourly`, `daily`, `weekly`, `monthly`, where all off could mean no report streams (and possibly help for backwards compat.?)

In `SourceBingAds` class, we'd then need to run some logic checking which of those toggles are on/off and setting up relevant report streams for each. e.g.
`hourly` and `monthly` are toggled **on**, so we add streams:
- AccountPerformanceReportHourly
- AccountPerformanceReportMonthly
- AdGroupPerformanceReportHourly
- AdGroupPerformanceReportMonthly
- and so on and so on for all reports...

This would give the user ultimate flexibility and _shouldn't_ require any major changes to the code, wdyt?